tasks.py

```diff
@@ -83,6 +83,36 @@ def lint(c):
     c.run('isort -c --recursive orion tests')
 
 
+@task
+def checkdeps(c, path):
+    with open('setup.py', 'r') as setup_py:
+        lines = setup_py.read().splitlines()
+
+    packages = []
+    started = False
+    for line in lines:
+        if started:
+            if line == ']':
+                started = False
+                continue
+
+            line = line.strip()
+            if line.startswith('#') or not line: # ignore comment
+                continue
+                
+            line = re.split(r'>?=?<?', line)[0]
+            line = re.sub(r"""['",]""", '', line)
+            packages.append(line)
+
+        elif line.startswith('install_requires = ['):
+            started = True
+
+    c.run(
+        f'pip freeze | grep -v \"sintel-dev/Orion.git\" | '
+        f'grep -E \'{"|".join(packages)}\' > {path}'
```

Comment:
Wondering which options are better. (1) check all packages defined in setup.py; (2) defined several core packages and only monitor them

Reply:
Thanks for the review @dyuliu!

The dependency checker helps us track package versions that are compatible with the software, if a new version is released that breaks Orion, it should be pinned. 

With the help of the dep checker, it should reduce manual checks of suddenly finding that a package breaks it, that is why I prefer to track all packages needed by Orion.